[antonlee59] iP

antonlee59 opened the pull request:
# The Anton Bot
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html))

The Anton Bot frees your mind of having to remember things you need to do. It's, 

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

ALl you need to do is,
1. download it from [here](url).
2. double-click it.
3. add yout tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
### 
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

antonlee59 committed:
Level 1. Greet, Echo, Exit

antonlee59 committed:
Level 2. Add, List

antonlee59 committed:
Level 3. Mark as Done

antonlee59 committed:
Level 4. ToDos, Events, Deadlines

antonlee59 committed:
Level 5. Handle Errors

antonlee59 committed:
Level 6. Delete

SHni99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,145 @@
 public class Duke {
+    // Constants
+    final static String PARTITION = "*******************************************";
+    final static String EXIT_COMMAND = "bye";
```
It would be better if you could create an enum list that contains the commands so you need no declare them one by one

SHni99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,145 @@
+                // Partition in UI
+
+                // Handling Various Commnds
+                if (input_command.equals(EXIT_COMMAND)) {
```
Is using Switch-case a better option than If-else?

SHni99 reviewed (commented):
Overall, LGTM. However, the input text file for testing is missing probably due to incomplete tasks but the rest of the issue is to fix a few nits such as listing the commands in enum class as it can help to reduce the amount of code written. Improvements can be done on the design such as handling the ui by creating methods in a UI class, instead of directly printing inside the action method (i.e handMark() ). But then, this is just a heads up before you start on A-MoreOOP task. Great job completing up to Level-6!! 😃

antonlee59 committed:
Level-7. Save

antonlee59 committed:
Level 8. Dates and Times

antonlee59 committed:
Resolve Merge Conflicts

antonlee59 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Merging remote branch upstream/add-gradle-support

BagDownB3nny commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+
+    // Fields
+    private static List<Task> current_list = new ArrayList<>();
+    private static Path path;
```
Perhaps `current_list` should be in camelcase instead?

BagDownB3nny commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+
+    // Methods
+    private static void checkInput(String[] current_input_array) throws EmptyDescriptionException {
+        if (current_input_array.length < 2) {
```
The camelcase naming convention can be applied to the arguments of these functions as well

BagDownB3nny commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,233 @@
+                } else {
+                    throw new InvalidCommandException("");
+                }
+                System.out.println(PARTITION);
+            } catch (Exception e) {
+                // handle exception
+                System.out.println(e.getMessage());
+                System.out.println(PARTITION);
```
Would adding the `System.out.println(PARTITION)` lines to a new `finally` block look better?

BagDownB3nny commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    @Override
+    public String toString() {
```
Maybe separating this one return statement into 3 lines of code would make it more readable?

BagDownB3nny reviewed (commented):
Overall your code adhered to the coding standards well! LGTM with some nits

antonlee59 committed:
Created Ui, Storage, Parser, TaskList classes

antonlee59 committed:
Added FindCommand to search with a keyword

antonlee59 committed:
Added JavaDocs to most of the non-trivial methods.

antonlee59 committed:
Tweaked the code to comply with a coding standard

antonlee59 committed:
Merge branch 'branch-Level-9'

Merging to add the FindCommand

antonlee59 committed:
Resolved merge conflict between master and branch-A-JavaDoc

elvern18 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,46 @@
+package duke;
+
+import duke.command.*;
```
Import classes can be stated explicitly to meet coding standards.

elvern18 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,46 @@
+    protected LocalDate to;
+
+    /**
+     * Initialises the from and to field in LocalDate type.
```
@params and @throws should have more details as well to meet coding standards.
E.g. @param description Description of task

elvern18 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,52 @@
+
+    /**
+     * Indicates whether the task is completed.
+     * @return
```
@return should also have a short description of what the function is returning.

elvern18 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,17 @@
+package duke.command;
```
Very clean OOP of Command abstract class!

elvern18 reviewed (commented):
LGTM, except for a few minor tweaks!

owen-yap commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+                currentCommand.handleCommand(ui);
+                isExit = currentCommand.isExit();
+                ui.showPartition();
+            } catch (InvalidCommandException e) {
+
+            }
```
Any reason why this exception is ignored?

owen-yap commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,46 @@
+
+    @Override
+    public String toString() {
+        return String.format("[E]%s (from: %s to: %s)", super.toString(), this.from.format(DateTimeFormatter.ofPattern("MMM dd yyyy")), this.to.format(DateTimeFormatter.ofPattern("MMM dd yyyy")));
```
Maybe consider abstracting different parts of this statement so that it is more readable and follows the coding standards?

owen-yap commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,46 @@
+import duke.command.*;
+
+public class Parser {
+    final static String EXIT_COMMAND = "bye";
+    final static String LIST_COMMAND = "list";
+    final static String DELETE_COMMAND = "delete";
+    final static String MARK_COMMAND = "mark";
+    final static String UNMARK_COMMAND = "unmark";
+    final static String TODO_COMMAND = "todo";
+    final static String EVENT_COMMAND = "event";
+    final static String DEADLINE_COMMAND = "deadline";
+    final static String FIND_COMMAND = "find";
```
Is there a more elegant way to present this list of commands?

owen-yap commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,76 @@
+            while (sc.hasNext()) {
+                String[] command = sc.nextLine().split(",");
+                Task newTask;
+                switch (command[0]) {
+                    case "T":
+                        newTask = new Todo(command[2]);
+                        break;
+                    case "D":
+                        newTask = new Deadline(command[2], command[3]);
+                        break;
+                    case "E":
+                        newTask = new Event(command[2], command[3], command[4]);
+                        break;
+                    default:
+                        throw new InvalidCommandException("");
```
Consider indenting switch case statements on the same indentation level to comply with the coding standards.

owen-yap commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,26 @@
+    public void addTask(Task newTask, boolean onLoadIn) {
+        this.add(newTask);
+        if (!onLoadIn) {
+            System.out.println("Got it. I've added this task:");
+            System.out.println("  " + this.toString());
+            System.out.println(String.format("Now you have %d tasks in the list.", this.size()));
```
Maybe get the UI class to handle these print statements?

owen-yap commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,76 @@
+        } else {
+            path = Paths.get(System.getProperty("user.dir"), filePath);
+        }
+        file = new File(path.toUri());
```
Nice implementation of storage using File. Checkout FileInputStream / FileOutputStream which serializes the TaskList object so that you don't have to write so much code to process the data!

owen-yap reviewed (commented):
Code looks functional! Some cosmetic changes and alternative implementation proposed. Otherwise LGTM!

antonlee59 committed:
Refactor code to suit the implementation of GUI using JavaFX

antonlee59 committed:
Add assertion statements to various part of the code base

Add assert statements to the file and path fileds to ensure that they are not null before the file is being scanned

Add assert statements to the various images to ensure that they are not null or referencing a wrong image

antonlee59 committed:
Merge pull request #2 from antonlee59/branch-A-Assertions

Add assertion statements to various part of the code base

antonlee59 committed:
A-CodeQuality

Change code to comply with the code Quality

Change code in multiple areas to comply with the Single Level of Abstraction Principle (SLAP)

antonlee59 committed:
Merge pull request #3 from antonlee59/branch-A-CodeQuality

A-CodeQuality

antonlee59 committed:
C-FriendlierSyntax

antonlee59 committed:
Add title to the GUI

antonlee59 committed:
Update README.md

antonlee59 committed:
A-Release

antonlee59 committed:
Update user guide
